dbt_semantic_interfaces/protocols/metric.py

```diff
@@ -280,6 +280,12 @@ def input_metrics(self) -> Sequence[MetricInput]:
     def metadata(self) -> Optional[Metadata]:  # noqa: D
         pass
 
+    @property
+    @abstractmethod
+    def meta(self) -> Dict[str, Any]:
+        """The meta field can be used to set metadata for a resource."""
+        pass
+
```

Comment:
We should move this to be nested under a `config` property similar to what we're doing in the `SemanticModel`